src/components/Keyboard.js

```diff
@@ -0,0 +1,31 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+
+import RaisedButton from 'material-ui/RaisedButton';
+
+import FontIcon from 'material-ui/FontIcon';
+const volumeUpIcon = <FontIcon className="material-icons">volume_up</FontIcon>;
+
+class Keyboard extends PureComponent {
+  render() {
+    return (
+      <div className="keyboard">
+        <textarea ref={ref => { this.textarea = ref }} placeholder="Type some text"></textarea>
+        <RaisedButton
+          label={this.props.intl.formatMessage({ id: 'cboard.containers.Text.speak' })}
+          labelPosition="before"
+          icon={volumeUpIcon}
+          primary={true}
+          onTouchTap={(event) => { event.preventDefault(); this.props.speak(this.textarea.value) }}
```

Comment:
dont create arrow functions here a new function will be created at every render so if pure render is used in Raised buttton then it will be of no use.